src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorDiagnosticFactory.cs

```diff
@@ -364,6 +364,30 @@ public static RazorDiagnostic CreateParsing_UnexpectedIdentifier(SourceSpan loca
     public static RazorDiagnostic CreateParsing_DirectiveExpectsIdentifierOrExpression(SourceSpan location, string directiveName)
         => RazorDiagnostic.Create(Parsing_DirectiveExpectsIdentifierOrExpression, location, directiveName);
 
+    internal static readonly RazorDiagnosticDescriptor Parsing_DefineAndUndefNotAllowed =
+        new($"{DiagnosticPrefix}1042",
```

Comment:
These new diagnostics should have distinct error codes, but currently have the same code (1042).

We could probably add a test that goes over all static fields of this class via reflection and verifies the error codes are distinct to avoid this in the future.

Reply:
Good catch, thanks.

Reply:
Sigh, 3015 was defined twice.